cpp/dolfinx/common/IndexMap.cpp

```diff
@@ -481,15 +481,10 @@ IndexMap::create_submap(std::span<const std::int32_t> indices) const
 
   // --- Step 2: Send ghost indices to owning rank
 
-  // Build list of src ranks (ranks that own ghosts)
-  std::vector<int> src = this->owners();
-  std::sort(src.begin(), src.end());
-  src.erase(std::unique(src.begin(), src.end()), src.end());
-
-  // Determine destination ranks (ranks that ghost my indices), and sort
-  std::vector<int> dest
-      = dolfinx::MPI::compute_graph_edges_nbx(this->comm(), src);
-  std::sort(dest.begin(), dest.end());
+  // Get source ranks (ranks that own ghosts) and destination ranks
+  // (ranks that ghost my indices)
+  std::vector<int> src = this->src();
```

Comment:
Could these be const references to avoid a copy?

Reply:
Whoops, yep!